waiter/src/waiter/token.clj

```diff
@@ -516,48 +516,62 @@
-                             :run-as-user run-as-user
-                             :status http-403-forbidden
-                             :log-level :warn}))))
-        (let [existing-service-description-owner (get existing-token-metadata "owner")]
-          (if-not (str/blank? existing-service-description-owner)
-            (when-not (authz/manage-token? entitlement-manager authenticated-user token existing-token-metadata)
-              (throw (ex-info "Cannot change owner of token"
-                              {:existing-owner existing-service-description-owner
-                               :new-user owner
-                               :status http-403-forbidden
-                               :log-level :warn})))
-            (when-not (authz/run-as? entitlement-manager authenticated-user owner)
-              (throw (ex-info "Cannot create token as user"
+      (let [existing-editor (get existing-token-metadata "editor")
+            existing-owner (get existing-token-metadata "owner")
+            owner? (and existing-owner (authz/manage-token? entitlement-manager authenticated-user token existing-token-metadata))
+            editor? (and existing-editor (not owner?)  (authz/run-as? entitlement-manager authenticated-user existing-editor))]
+        (when editor?
+          (log/info "applying editor privileges to operation" {:editor authenticated-user :owner existing-owner}))
+        (if editor?
+          (let [existing-token-parameters (sd/token->token-parameters kv-store token :include-deleted false)]
+            (doseq [parameter-name ["editor" "owner" "run-as-user"]]
+              (let [existing-value (get existing-token-parameters parameter-name)
+                    new-value (get new-token-data parameter-name)]
+                (when (not= existing-value new-value)
+                  (throw (ex-info (str "Not allowed to edit parameter " parameter-name)
+                                  {:authenticated-user authenticated-user
+                                   :existing-token-description existing-token-parameters
+                                   :parameter parameter-name
+                                   :parameter-exiting-value existing-value
+                                   :parameter-new-value new-value
+                                   :privileges {:editor? editor? :owner? owner?}
+                                   :status http-403-forbidden
+                                   :log-level :warn}))))))
+          ;; only check run-as-user rules when not running as editor
+          (when (and run-as-user (not= "*" run-as-user))
+            (when-not (authz/run-as? entitlement-manager authenticated-user run-as-user)
+              (throw (ex-info "Cannot run as user"
                               {:authenticated-user authenticated-user
-                               :owner owner
+                               :privileges {:editor? editor? :owner? owner?}
```

Comment:
Can `editor?` ever be anything other than `false` here?  Should we revert back to the old log message?